README.md

```diff
@@ -71,6 +72,146 @@ operations of generating and manipulating entity graphs.
 
 ## Tutorial
 
+In learning any new tool, I think it's useful to begin by learning the
+tool's purpose, then getting a high-level overview of the architecture
+and how it achieve's the tool's purpose. With those concepts in place,
+concrete examples and exercises will help you understand how to use
+the tool. So let's start with Specmonstah's purpose.
+
+Specmonstah was built to aid testing by generating and inserting
+records in a database in dependency order. For example, if you want to
+test the insertion of a Todo, but that Todo depends on the existence
+of a TodoList, and the TodoList depends on a User, then Specmonstah
+will generate User and TodoList records and insert them without your
+having to clutter your test with code related to Users and TodoLists.
+
+To accomplish this, Specmonstah does the following:
+
+**1. Graph generation (and ent/ent type definitions)**
+
+It generates a graph whose nodes correspond to _entity types_ (or _ent
+types_) and _entities_ (or just _ents_):
+
+![Simple todo example](docs/todo-example.png)
+
+In the graph above, the `:todo`, `:todo-list`, and `:user` nodes
+correspond to ent types, and the rest correspond to ents.
+
+We're going to be using the terms _ent_ and _ent type_ a lot, and
+you'll see them all throughout the source code, so let's define them:
+
+**Ent type.** An ent type is kind of analogous to a relation in a
+relational database, or a class in object-oriented programming. It
+differs in that relations and classes define all the attributes of
+their instances, whereas ent types don't. Ent types define how ent
+instances are related to each other. For example, the `:todo` ent type
+wouldn't include a `:description` attribute, but it does specify that
+a `:todo` instances reference `:todo-list` instances. In the next
+section you'll learn how to define ent types.
+
+Ent types are represented as nodes in the Specmonstah graph (let's
+abbreviate that with _SG_). Ent types have directed edges to their
+instances. It's rare that you'll interact with ent types directly.
+
+**Ent.** An ent is an instance of an ent type. They have a name (`:t0,
+:u0`, etc), and reference other ents. They're represented as nodes in
+the SG, with directed edges going from ents to the ents they
+reference; there's a directed edge from `:tl0` to `:u0` because `:tl0`
+references `:u0`.
+
+Ents can be associated with additional data - for example, a user ent
+can be associated with a map of user data generated by
```

Comment:
Maybe something like "a map of user data, like name and email" so it's totally clear that you're not talking about relation data.